kernel/include/kernel_tls.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Kernel Thread Local Storage APIs.
+ *
+ * Kernel APIs related to thread local storage.
+ */
+
+#ifndef ZEPHYR_KERNEL_INCLUDE_KERNEL_TLS_H_
+#define ZEPHYR_KERNEL_INCLUDE_KERNEL_TLS_H_
+
+#include <linker/linker-defs.h>
+
+/**
+ * @brief Return the total size of TLS data/bss areas
+ *
+ * This returns the total size of thread local storage (TLS)
+ * data and bss areas as defined in the linker script.
+ * Note that this does not include any architecture specific
+ * bits required for proper functionality of TLS.
+ *
+ * @return Total size of TLS data/bss areas
+ */
+static inline size_t z_tls_data_size(void)
+{
+	return (size_t)(__tls_end - __tls_start);
+}
```

Comment:
As you can see, providing link-time computed size values will simplify this code.